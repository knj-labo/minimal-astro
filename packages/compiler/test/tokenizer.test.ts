import { describe, expect, it } from 'vitest'
import { tokenize } from '../src/tokenizer/index.js'

describe('Tokenizer', () => {
  it('should tokenize simple HTML', () => {
    const source = '<div>Hello World</div>'
    const tokens = tokenize(source)

    expect(tokens).toMatchSnapshot()
  })

  it('should tokenize frontmatter', () => {
    const source = `---
const title = 'Hello'
---`
    const tokens = tokenize(source)

    expect(tokens).toMatchSnapshot()
  })
})
