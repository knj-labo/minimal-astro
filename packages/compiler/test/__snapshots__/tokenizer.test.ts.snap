// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tokenizer > should tokenize frontmatter 1`] = `
[
  {
    "col": 1,
    "line": 1,
    "type": "FRONTMATTER_START",
    "value": "---",
  },
  {
    "col": 22,
    "line": 2,
    "type": "FRONTMATTER_CONTENT",
    "value": "const title = 'Hello'",
  },
  {
    "col": 1,
    "line": 3,
    "type": "FRONTMATTER_END",
    "value": "---",
  },
  {
    "col": 4,
    "line": 3,
    "type": "EOF",
    "value": "",
  },
]
`;

exports[`Tokenizer > should tokenize simple HTML 1`] = `
[
  {
    "col": 1,
    "line": 1,
    "type": "HTML_TAG_OPEN",
    "value": "<",
  },
  {
    "col": 5,
    "line": 1,
    "type": "HTML_TAG_NAME",
    "value": "div",
  },
  {
    "col": 5,
    "line": 1,
    "type": "HTML_TAG_CLOSE",
    "value": ">",
  },
  {
    "col": 17,
    "line": 1,
    "type": "TEXT",
    "value": "Hello World",
  },
  {
    "col": 17,
    "line": 1,
    "type": "HTML_TAG_CLOSE",
    "value": "</",
  },
  {
    "col": 22,
    "line": 1,
    "type": "HTML_TAG_NAME",
    "value": "div",
  },
  {
    "col": 22,
    "line": 1,
    "type": "HTML_TAG_CLOSE",
    "value": ">",
  },
  {
    "col": 23,
    "line": 1,
    "type": "EOF",
    "value": "",
  },
]
`;
